plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -168,22 +168,22 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sql_query += query_addon
 
 		if p.Query[i].Version <= db_version {
-			rows, err := db.Query(sql_query)
+			rows, err := p.DB.Query(sql_query)
```

Comment:
maxlifetime solves the issue (of a connection being terminated by the db due to idle time) by forcing the driver to not reuse a connection after its been open for longer than the maxlifetime.

>First you'd have to start a transaction to ensure that the query uses the same connection the ping just used

we've already established that these plugins have only ever used one connection (this PR makes no changes to that fact). Transaction isnt necessary. 

You've basically complained that
1) the connection may have been killed by a internal db process. Fine, ping actually will force it to reconnect by triggering that extra query and has a retry builtin. this prevents the first metric query from failing due to that particular situation, which could also be solved by setting maxlifetime to a value lower than the idle connection timeout.
2) now you're complaining about the latency of adding the ping.

so either 1) accept that manually setting the max lifetime to a lower value is a solution to that. or 2) accept the added latency of the ping as a builtin protection against the issue.

I'm not interested in doing retry logic for such rare cases which already have 2 solutions builtin to this pr. yes a bunch of shit can go wrong at anytime none of which was being handle prior to this PR anyways, and quite frankly in practice has no real impact. Feel free to write your own PR implementing whatever crazy retry logic you want.

regarding the examples:
1. is referring to a succesful ping after a failed query which makes perfect sense because if you look at the sql.DB code it discards a connection when a broken error is detected.

2. actually discusses the exact use case its being put to use for here. if something goes wrong between the ping/first query that is literally no different than something going wrong between any of the queries and is outside the scope of this PR.